### PR TITLE
Rewrite Component mam with mamv2 details

### PR DESCRIPTION
**Pull request by jkrvivian**

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -206,11 +206,23 @@ \subsubsection{Participants}
...
 
-One can create multiple channels, however, the size of a channel is fixed which is decided before creation. Thus data providers need to firstly decide how to distribute data product into MAM channels. While the creation of a MAM channel is time-consuming, brokers are also responsible for channel creation, encrypted data publishing and session key certification in our system. See Figure \ref{fig:launching_product}
+Either channel or endpoint are like singly linked-list. The MAM channel's ID and endpoint's ID are the addresses of transaction and are the root of a channel and endpoint. Each address of a message can be derived from the previous one. A user can create multiple channels and multiple endpoints under the same channel. The messages can be encrypted with the session key and broadcasted in chronological order to the Tangle by attaching it to an endpoint. This allows only entities that know the session key can decode these messages after retrieving it from the Tangle, and also implements the concept of forward secrecy, where one have no access to the data back from his/hers entry point. Fig. shows the architecture of MAM.
```
Comment (twzjwang) [inline paper.tex]: Either channel or endpoint "is"

Comment (jserv) [reply paper.tex]: Should be "singly-linked list", not "singly linked-list"

**Comment by jkrvivian** [comment]
* Interaction between provider and MAM
* Why MAM
* Efficiency problem of MAM

These 3 parts are added.

File: paper.tex [hunk 2]
```diff
@@ -206,12 +206,29 @@ \subsubsection{Participants}
...
+
+% why MAM
+MAM is considered to be the most important fundamental building block in the proposed system. The decentralized and fault-tolerant charactaristic of distributed ledgers reduce the risks of centralized storage services, and the underlying IOTA network is scalable which withstands real-time data while increasing users all over the world. Moreover, the features of MAM makes it an even better data storage which allows preserving streaming data while ensuring data integrity, proving data ownership to any participants and managing session keys of data. And through the access control and the desing of channel, endpoint of MAM, participants are allowed to subscribe the future data.
```
Comment (jserv) [inline paper.tex]: Change "considered to be" to "considered as."

File: paper.tex [hunk 3]
```diff
@@ -206,12 +206,29 @@ \subsubsection{Participants}
...
+
+% interaction between provider
+Data provider is the main role to interact with MAM. With a seed, a provider can starts creating a channel or endpoint. As above mentioned, the size of a channel and an endpoint is fixed, thus providers need to decide the selling unit of data products according to its data type or pricing strategy. Also data providers can a make product preview which is public and not encrypted on MAM. In Figure \ref{fig:mam_struct}, a "central endpoint" means its endpoint's ID is the same as channel's ID. By attaching part of the product to central endpoint can give consumers a quick preview with channel's ID only. Then if the trade establishes, providers can then give consumers the encrypted endpoint and session key to consumers. After retrieving the messages on endpoints, users can easily verify the content with digital signatures included in messages.
```
Comment (jserv) [inline paper.tex]: Change "can starts " to "can start."

Comment (jserv) [reply paper.tex]: Change "the size of" to "the length of" or "the capacity of."

File: references.bib [hunk 4]
```diff
@@ -74,6 +74,24 @@ @misc{MAM
...
 
+@misc{MAMSpec,
+  title = {Masked Authenticated Messaging Version 2 Protocol},
+  url = {https://github.com/iotaledger/entangled/blob/develop/mam/spec.pdf},
+  urldate = {2019-03-07}
+}
+
+@article{MAMDescription,
+  title = "Authenticating Health Activity Data Using Distributed
+Ledger Technologies",
```
Comment (jserv) [inline references.bib]: Don't wrap into two lines.

**Pull request merged**